### ISSUE TITLE
`use_key_in_widget_constructors` should be reported at the identifier of the constant constructors

### ISSUE THREAD

**Issue by asashour** [dart-lang/linter#3368]
With a case like:

```dart
import 'package:flutter/material.dart';

class MyWidget extends StatelessWidget {
  const MyWidget();

  @override
  Widget build(BuildContext context) {
    return Container();
  }
}
```

there is no suggested fix for the lint.

The reason seems to be that the lint is reported at the `ConstructorDeclaration`, however the fix is expecting the `parent` to be a `ConstructorDeclaration`.

https://github.com/dart-lang/sdk/blob/dd007c2c71978996502dee1c582e277389b33969/pkg/analysis_server/lib/src/services/correction/dart/add_key_to_constructors.dart#L52-L54

I think the lint should be changed in this case, instead of the fix. In which the lint reports at the `SimpleIdentifier`, not the `ConstructorDeclaration`.

**Comment by bwilkerson** [comment]
I agree. In general, diagnostics should be reported on the smallest unit of code that points users to the source of the issue (typically a single token). If we're reporting a diagnostic against a whole constructor declaration, then that should be changed, and the convention we follow is to highlight the name being declared rather than the whole declaration.
